File: examples/anthropic-managed-agents/python/anthropic_managed_agents_e2b/agent.py
from __future__ import annotations

from collections.abc import Sequence

import anthropic

SANDBOX_TOOLS = ("bash", "read", "write", "edit", "glob", "grep")
WEB_TOOLS = ("web_fetch", "web_search")
DEFAULT_MODEL = "claude-sonnet-4-6"
DEFAULT_SYSTEM_PROMPT = "You have a Linux sandbox. Use the available tools to complete the task."


def create_agent(
    *,
    api_key: str,
    name: str,
    model: str = DEFAULT_MODEL,
    sandbox_tools: Sequence[str] = SANDBOX_TOOLS,
    web_tools: Sequence[str] = WEB_TOOLS,
):
    client = anthropic.Anthropic(api_key=api_key)
    return client.beta.agents.create(
        name=name,
        model=model,
        system=DEFAULT_SYSTEM_PROMPT,
        tools=[
            {
                "type": "agent_toolset_20260401",
                "default_config": {
                    "enabled": False,
                    "permission_policy": {"type": "always_allow"},
                },
                "configs": [
                    {
                        "name": tool,
                        "enabled": True,
                        "permission_policy": {"type": "always_allow"},
                    }
                    for tool in (*sandbox_tools, *web_tools)
                ],
            }
        ],
    )

File: examples/anthropic-managed-agents/javascript/src/agent.ts
import Anthropic from "@anthropic-ai/sdk";

import {
  DEFAULT_MODEL,
  DEFAULT_SYSTEM_PROMPT,
  SANDBOX_TOOLS,
  WEB_TOOLS,
} from "./constants.js";

type SandboxTool = (typeof SANDBOX_TOOLS)[number];
type WebTool = (typeof WEB_TOOLS)[number];
type ManagedAgentTool = SandboxTool | WebTool;

export async function createAgent({
  apiKey,
  name,
  model = DEFAULT_MODEL,
  sandboxTools = SANDBOX_TOOLS,
  webTools = WEB_TOOLS,
}: {
  apiKey: string;
  name: string;
  model?: string;
  sandboxTools?: readonly SandboxTool[];
  webTools?: readonly WebTool[];
}) {
  const client = new Anthropic({ apiKey });
  const enabledTools: ManagedAgentTool[] = [...sandboxTools, ...webTools];
  return client.beta.agents.create({
    name,
    model,
    system: DEFAULT_SYSTEM_PROMPT,
    tools: [
      {
        type: "agent_toolset_20260401",
        default_config: {
          enabled: false,
          permission_policy: { type: "always_allow" },
        },
        configs: enabledTools.map((tool) => ({
          name: tool,
          enabled: true,
          permission_policy: { type: "always_allow" },
        })),
      },
    ],
  });
}
